### ISSUE TITLE
Improve Server Information menu readability (make hostname+port easier to find)

### ISSUE THREAD

**Issue by SeerLite** [mumble-voip/mumble#4773]
**Context**
Stuff like the server host name and port are not intuitive to find at all. Right now I find it easier to open the Connect menu and click Edit on the server to get the information, while the actual Information menu should show this a lot clearer.

**Describe the feature you have in mind**
Improve the readability of the Information menu/dialog. Show information important to the user at the top and debugging info at the bottom. Things like host name, port, version and maybe connected users should be visible to the user immediately. Stuff like the encryption used and TLS version aren't things most users will be looking for.
Basically what I'm asking for is moving the hostname and port from the "Control channel" section to the top of the dialog/menu/thing.

**Describe alternatives you've considered**
Make each section a key-value table so it's easier to find whatever is important no matter its location.

**Comment by Krzmbrzl** [comment]
For reference, this is how the window currently looks like:
![image](https://user-images.githubusercontent.com/12751591/108231513-22c44380-7142-11eb-9f1b-2f227c81c636.png)


I agree that this does not put emphasis on the things most users are probably interested in.

**Comment by Krzmbrzl** [comment]
@SeerLite any chance you would want to work on this yourself? It should be quite straight forward. I'd also assist you in that, if you want me to.

**Comment by SeerLite** [comment]
Hey sure, I can give it a try! I had a look and it doesn't look that complicated. Just a question though: How should I handle the headers/section names? I'm thinking of moving the general information from the "Control channel" section to a new one at the top called "General" or "General information". Would that be fine?

Also now that I've been staring at the window for a while I think it may be too tall(?) Would it be too difficult to add a scrollbar and make the window a bit wider and shorter too?

**Comment by Krzmbrzl** [comment]
Cool :+1: 

I think just moving entries around will make the categories inconsistent. Right now they do follow a pattern, that can be understood. So I think the idea should be to change this pattern so that it better fits the average user.

I think I would probably suggest something like this:
- Server information (protocol version, OS, URL, port, users)
- Audio bandwidth (as is)
- Connection details (split for TCP (Control) and UDP (Voice) - encryption, latency, TLS version, encryption method)

I think the UDP statistics should then be in the UDP section of the connection details.

The names are of course not set in stone. Using "General" or "General information" for the server info category sounds like a good choice as well (maybe even better since the entire window is named "Server information").

> Also now that I've been staring at the window for a while I think it may be too tall(?) Would it be too difficult to add a scrollbar and make the window a bit wider and shorter too?

In my experience adding scrollbars usually ends up creating a bunch of complications that need to be dealt with. It's definitely doable but usually it takes me longer than I originally expected. This could also be because I am not regularly programming GUI stuff :shrug: 

In general though I think if we wanted to make the window shorter, I think a very elegant thing to do would be to make the sections collapsible. A quick search online revealed that this is probably quite some work as there doesn't seem to be standard Qt widgets that can do this for us.

**Comment by SeerLite** [comment]
Hi! Sorry, something came up so I couldn't look into this before. Now I have some time to check it out :p

> * Server information (protocol version, OS, URL, port, users)

I've reordered it a bit to fit what I as an user think is more important:
![image](https://user-images.githubusercontent.com/21978144/109357505-0ff7f000-7861-11eb-837f-3548b0f74e5c.png)
What do you think?

> * Connection details (split for TCP (Control) and UDP (Voice) - encryption, latency, TLS version, encryption method)

I'm having some trouble understanding this. Should it be a section (`<h2>`) called "Connection details" with subsections (`<h3>`) for TCP and UDP?


About the shorter window/scrollbar: I think having a scrollbar is better than collapsible sections, I feel like it would be a much neater experience. IMO the collapsible sections feels off for something like this (information), where you usually want to see everything at once.

I understand it seems like a lot of work though (and I wouldn't even know where to start haha) so I guess I'll make a PR for reordering the info first and then see what can be done about the window size.

**Comment by SeerLite** [comment]
I also just realized there's a whole translation system and lots of supported languages 0_0 How should I handle that?

**Comment by Krzmbrzl** [comment]
> What do you think?

The order is looking good :+1: 
There are a few cosmetic things that I think could be improved though:
1. The host name and the port are not aligned (the port is listed slightly more to the left as the corresponding label is smaller). I think for this we should use a `QFormLayout`. That will make sure that the alignment works out.
2. I think `Protocol` should be followed by a colon (aka `Protocol:`) as this is how the other information is presented as well
3. The OS information is floating there without any label explaining what this is supposed to mean. I propose using a format like `Server <release> running on <OS> <OS version>`.

I also had a look at the source code and realized that atm this entire window's content is created as a single String. Imo it'd be best to create a dedicated UI component for this. Using Qt Designer it is rather straight forward to create one. Do you think you could do that? Or does that go too far?
(This is probably also what is required to properly integrate scrollbar anyways)

> I'm having some trouble understanding this. Should it be a section (<h2>) called "Connection details" with subsections (<h3>) for TCP and UDP?

Yes

> About the shorter window/scrollbar: I think having a scrollbar is better than collapsible sections, I feel like it would be a much neater experience. IMO the collapsible sections feels off for something like this (information), where you usually want to see everything at once.
>
> I understand it seems like a lot of work though (and I wouldn't even know where to start haha) so I guess I'll make a PR for reordering the info first and then see what can be done about the window size.

Sounds good :)

> I also just realized there's a whole translation system and lots of supported languages 0_0 How should I handle that?

As long as you are always using the `tr(...)` function to create the Strings, translation is taken care of (tr = translate).

**Comment by SeerLite** [comment]
Hi! Sorry again for taking so long to respond!

I fixed 2 and 3, but I don't know how to do the `QFormLayout` thing. I have very little experience in C/C++. I tried to do it a while back and also tried to use Qt Designer, but I couldn't figure either it out so I postponed it (and then forgot about it haha sorry!). Also it looks like `QFormLayout `is more like for forms and input, are you sure it's the right thing for this? It doesn't seem to accept two `QString`s as row values, only one and it's for a label. 

> As long as you are always using the `tr(...)` function to create the Strings, translation is taken care of (tr = translate).

Do I not have to do anything else besides just using `tr`? Like adding the actual translations and stuff?

---
I opened #4887

**Comment by Krzmbrzl** [comment]
> Also it looks like QFormLayout is more like for forms and input, are you sure it's the right thing for this? It doesn't seem to accept two QStrings as row values, only one and it's for a label.

Yes I am sure that this is what is suited for this job. The layout doesn't care about what exactly it is that it is positioning but the gist of a form layout is that you have some elements to the left (typically right-aligned) which take up only as much space as absolutely necessary whereas next to these is another element that is stretched to fill the remaining space. This is typically used for label + input field as used in forms (thus the name) but it works equally well for label/label combinations for read-only data display.

Yes a layout does not work on Strings. That's why I suggested to forget about the current String approach and building a proper UI instead (with individual widgets which is what a layout operates on).

> Do I not have to do anything else besides just using tr? Like adding the actual translations and stuff?

No. Using `tr` will mark the String as translatable and the actual process of creating the translations for it happens via Weblate.

**Comment by Kissaki** [comment]
As this is being touched, I think rather than a QLabel which is not text-selectable, it would be nice if the hostname+port could be copied; potentially with a button, raw or as a mumble URL, or solved as selectable text.

**Comment by Krzmbrzl** [comment]
labels can be made selectable. And that's what I did in my current working draft :)
